File: ProgLab1/PyLab1/PyLab1/MyFile.py
def GetChangedFile(filePath):
    lines = []
    changedText = []
    currentLine = ""
    numbersAmount = 0
    counter = 0

    file = open(filePath, 'r')

    lines = file.readlines()
    for i in range(0, len(lines)):
        numbersAmount = 0
        currentLine = lines[i].replace("\n", "")
        for j in range(0, len(currentLine)):
            if(currentLine[j].isdigit()):
                numbersAmount += 1

        if(i % 2 == 0):
            currentLine += " Digits amount: " + str(numbersAmount)
        changedText.append(currentLine)
    return changedText

def FillFile(text, filePath):
    file = open(filePath, 'w')
    for i in range(0, len(text)):
        file.write(text[i] + "\n")

def Display(filePath, caption = "Original file"):
    print(caption)

    file = open(filePath, 'r')
    lines = file.readlines()
    for i in range(0, len(lines)):
        print(lines[i].replace("\n", ""))

File: ProgLab1/Lab1/Lab1/MyFile.cs
using System;
using System.Collections.Generic;
using System.IO;

namespace FileOperations
{
    public class MyFile
    {
        public static List<string> GetChangedFile(StreamReader sr)
        {
            List<string> changedText = new List<string>();
            string line;
            int numbersAmount;

            for (int i = 0; (line = sr.ReadLine()) != null; i++)
            {
                numbersAmount = 0;

                for (int j = 0; j < line.Length; j++)
                    if (Char.IsNumber(line[j]))
                        numbersAmount++;

                if (i % 2 == 0)
                    line += " Digits amount: " + numbersAmount;

                changedText.Add(line);
            }
            return changedText;
        }

        public static void Fill(List<string> text, StreamWriter sw)
        {
            for (int i = 0; i < text.Count; i++)
            {
                sw.WriteLine(text[i]);
            }
        }

        public static void Display(StreamReader sr, string caption = "Original file")
        {
            sr.BaseStream.Position = 0;
            Console.WriteLine(caption);
            Console.WriteLine(sr.ReadToEnd());
        }
    }
}

